Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitImmersiveMode.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2025 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+
+#include "ConsoleMessageTest.h"
+#include "WebKitTestServer.h"
+
+static const char indexHTML[] =
+"<!DOCTYPE html><html><body>"
+"<button id='enterXR'>Enter Experience</button>"
+"<script>"
+"document.getElementById('enterXR').addEventListener('click', () => {"
+"  navigator.xr.requestSession('immersive-vr').then(session => {"
+"    console.log('XR session started');"
+"    session.addEventListener('end', (event) => {"
+"        console.log('XR session ended');"
+"    });"
+"  }).catch(err => console.error(`XR session failed to start: ${err}`));"
+"});"
+"</script></body></html>";
+
+class ImmersiveModeTest : public ConsoleMessageTest {
```

Comment:
I don think it's a good idea to do this, this is not a console message test. Instead of using console messages you can use user script messages. In other tests we also use the title for simple cases like this.